src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -145,10 +150,9 @@ public LockFile CreateLockFile(LockFile previousLockFile,
                                 && StringComparer.Ordinal.Equals(path, previousLibrary.Path)
                                 && StringComparer.Ordinal.Equals(sha512, previousLibrary.Sha512))
                             {
-                                // We mutate this previous library so we must take a clone of it. This is
```

Comment:
I don't understand why we needed to clone here since we don't seem to be mutating the lock file library at all.

Reply:
I wasn't clear on that either. I modified the comment, but didn't really look much further into it.